cartography/intel/kubernetes/ingresses.py

```diff
@@ -0,0 +1,101 @@
+import logging
+from typing import Dict
+from typing import List
+
+from neo4j import Session
+
+from cartography.intel.kubernetes.util import get_epoch
+from cartography.intel.kubernetes.util import K8sClient
+from cartography.util import timeit
+
+logger = logging.getLogger(__name__)
+
+
+@timeit
+def sync_ingresses(
+    session: Session,
+    client: K8sClient,
+    update_tag: int,
+    cluster: Dict,
+) -> List[Dict]:
+    ingresses = get_ingresses(client, cluster)
+    load_ingresses(session, ingresses, update_tag)
+    return ingresses
+
+
+@timeit
+def get_ingresses(client: K8sClient, cluster: Dict) -> List[Dict]:
+    return [
+        {
+            "uid": ingress.metadata.uid,
+            "name": ingress.metadata.name,
+            "creation_timestamp": get_epoch(ingress.metadata.creation_timestamp),
+            "deletion_timestamp": get_epoch(ingress.metadata.deletion_timestamp),
+            "namespace": ingress.metadata.namespace,
+            "cluster_uid": cluster["uid"],
+            "labels": ingress.metadata.labels,
+            "rules": [
+                {
+                    "host": r.host,
+                    "http_paths": [
+                        {
+                            "path": p.path or "",
+                            "path_type": p.path_type if p.path else "Empty",
+                            "service": p.backend.service.name,
+                            "port": p.backend.service.port.number,
+                        }
+                        for p in r.http.paths
+                    ],
+                }
+                for r in ingress.spec.rules
+            ],
+            "tls": ingress.spec.tls,
+        }
+        for ingress in client.networking.list_ingress_for_all_namespaces().items
```

Comment:
We usually like to have `get_`, `transform_`, and `load_` in different functions.  Here, you combine the get and transform. I didn't correct this in your previous PR about KubernetesSecrets, but try to follow the pattern in this PR.

Reply:
I have not found a `transform_` function in the `kubernetes` module (I used other submodules as templates for my code).

Do you think I could do that in another PR? Maybe with other `kubernetes` submodules if appropriate?

Reply:
The `get_` is meant only to do the api fetch. And the `transform_` is meant to munge your original list of dictionaries into another list of dictionaries that your `load_` will use.

Reply:
Yes I totally understand. What I meant is: this pattern seems not in use yet in the whole k8s submodule, so if you agree, I could do that in another PR, for all the components at once.

Reply:
I know it's not used in this module, but it really should be used.

Reply:
That's done. Does that match what you expect?